faiss.yaml

```
@@ -0,0 +1,54 @@
+package:
+  name: faiss
+  version: 1.10.0
+  epoch: 0
+  description: A library for efficient similarity search and clustering of dense vectors.
+  copyright:
+    - license: MIT
+
+environment:
+  contents:
+    packages:
+      - build-base
+      - busybox
+      - cmake
+      - gflags-dev
+      - openblas-dev
+
+pipeline:
+  - uses: git-checkout
+    with:
+      destination: ${{package.name}}
+      expected-commit: 189a9d4461233de798c2eb17fb6d354ceda542e6
+      repository: https://github.com/facebookresearch/faiss
+      tag: v${{package.version}}
+
+subpackages:
+  - name: ${{package.name}}-cpu
+    description: ${{package.name}} for CPU only
+    pipeline:
+      - uses: cmake/configure
+        working-directory: ${{package.name}}
+        with:
+          opts: |
+            -DBUILD_SHARED_LIBS=ON \
+            -DBUILD_TESTING=OFF \
+            -DFAISS_ENABLE_C_API=ON \
+            -DFAISS_ENABLE_GPU=OFF \
+            -DFAISS_ENABLE_PYTHON=OFF
+      - uses: cmake/build
+        working-directory: ${{package.name}}
+      - uses: cmake/install
+        working-directory: ${{package.name}}
```

nit: not-a-blocker: you should be able to do something like:
```suggestion
      - working-directory: ${{package.name}}
        pipeline:
        - uses: cmake/configure
          with:
            opts: |
              -DBUILD_SHARED_LIBS=ON \
              -DBUILD_TESTING=OFF \
              -DFAISS_ENABLE_C_API=ON \
              -DFAISS_ENABLE_GPU=OFF \
              -DFAISS_ENABLE_PYTHON=OFF
      - uses: cmake/build
      - uses: cmake/install
```